app/src/main/java/com/example/cinemapp/data/MovieLocalCache.kt

```diff
@@ -3,41 +3,120 @@ package com.example.cinemapp.data
 import android.util.Log
 import com.example.cinemapp.data.model.CastMovieCreditDTO
 import com.example.cinemapp.data.model.MovieDTO
+import com.example.cinemapp.data.model.MovieResponseDTO
 import com.example.cinemapp.data.model.PersonMovieCreditsResponseDTO
 
 class MovieLocalCache {
 
     private val upcomingCache = mutableMapOf<Int, ArrayList<MovieDTO>>()
+    private var upcomingPages: Int? = 1
```

Comment:
Naming indicates more than one Int here. Should this variable be named something like `currentPageUpcomingMovies`?

Reply:
It means the total number of upcoming pages that exist.
I store it there so when i want to get a movie list i also know the total amount of pages that can be loaded.
This is used to know when to stop trying to load them when scrolling.

Reply:
Changed it to upcomingPagesTotal.